app/code/Magento/ConfigurableProduct/Plugin/Model/ResourceModel/Product.php

```diff
@@ -21,18 +28,42 @@ class Product
      */
     private $productIndexer;
 
+    /**
+     * @var ProductAttributeRepositoryInterface
+     */
+    private $productAttributeRepository;
+
+    /**
+     * @var \Magento\Framework\Api\SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var \Magento\Framework\Api\FilterBuilder
+     */
+    private $filterBuilder;
+
     /**
      * Initialize Product dependencies.
      *
      * @param Configurable $configurable
      * @param ActionInterface $productIndexer
+     * @param ProductAttributeRepositoryInterface $productAttributeRepository
+     * @param \Magento\Framework\Api\SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param \Magento\Framework\Api\FilterBuilder $filterBuilder
      */
     public function __construct(
         Configurable $configurable,
-        ActionInterface $productIndexer
+        ActionInterface $productIndexer,
+        ProductAttributeRepositoryInterface $productAttributeRepository,
```

Comment:
Please use `ProductAttributeRepositoryInterface $productAttributeRepository = null` and `$this->productAttributeRepository = $productAttributeRepository ?: ObjectManager::getInstance()->get(ProductAttributeRepositoryInterface::class)` for backward compatible

Reply:
Similar other classes